data/calculated_parser/parser.py

```diff
@@ -69,8 +66,36 @@ def get_key_for_variable(self, variable):
                 key = [d[k] for k in variable.dimensions]
         except KeyError:
             raise SyntaxError
+        return tuple(key)
+
+    def get_key_for_var_all(self, variable):
```

Comment:
A docstring would be suitable here I think.

Reply:
Updated docstring, can you take a look and let me know if it's sufficient.
I just referenced the above function and made note of the difference (doesn't perform depth slicing)